src/Compiler/pars.fsy

```diff
@@ -168,6 +168,7 @@ let parse_error_rich = Some(fun (ctxt: ParseErrorContext<_>) ->
 %type <SynExpr> declExprBlock
 %type <SynPat> headBindingPattern
 %type <SynType> atomTypeNonAtomicDeprecated
+%type <SynType> atomTypeOrAnonRecdType
```

Comment:
For my understanding, why this precedence was added? As in, what wouldn't be working without it?

Reply:
> For my understanding, why this precedence was added?

It's not a precedence, it's a type annotation. Without it the access to `Range` property didn't work.

Reply:
Right, looked at the wrong place. Okay